spec/features/adding_items_to_cart_spec.rb

```diff
@@ -155,6 +155,130 @@
         end
       end
     end
+
+    context "when both of the bundle items have a user-selectable variant", js: true do
+      let(:bundle) { create(:product_in_stock, name: "Bundle", sku: "BUNDLE") }
+      let(:keychain) do
+        create(:product_in_stock, name: "Keychain",
+                                  sku: "KEYCHAIN",
+                                  can_be_part: true)
+      end
+      let(:shirt) do
+        shirt, _shirts_by_size = create_bundle_product_with_options(
+          name: "Shirt",
+          sku: "SHIRT",
+          option_type: "Size",
+          option_values: ["Large", "XL"]
+        )
+
+        shirt
+      end
+      let(:hat) do
+        hat, _hats_by_size = create_bundle_product_with_options(
+          name: "Hat",
+          sku: "HAT",
+          option_type: "Color",
+          option_values: ["Red", "Blue"]
+        )
+
+        hat
+      end
+
+      before do
+        add_part_to_bundle(
+          bundle,
+          keychain.master,
+          count: 1
+        )
+
+        add_part_to_bundle(
+          bundle,
+          shirt.master,
+          variant_selection_deferred: true
+        )
+
+        add_part_to_bundle(
+          bundle,
+          hat.master,
+          variant_selection_deferred: true
+        )
+      end
+
+      context "and the user selects differing variants from the existing line item" do
+        specify "the cart contains incremented variants when listing bundle items" do
+
+          visit spree.product_path(bundle)
+
+          select "Size: Large", from: "options_selected_variants_2"
```

Comment:
I feel like these three lines can be refactored as a well named method as well. Maybe `add_item_to_cart(size: "Large", color: "Red")`. That way, if the field ever change later we don't have to change 8 places.